File: tusow/src/main/java/it/unibo/coordination/tusow/exceptions/HttpError.java
package it.unibo.coordination.tusow.exceptions;

public class HttpError extends RuntimeException {

    private final int statusCode;

    public HttpError(int statusCode) {
        this.statusCode = statusCode;
    }

    public HttpError(int statusCode, String message) {
        super(message);
        this.statusCode = statusCode;
    }

    public HttpError(int statusCode, String message, Throwable cause) {
        super(message, cause);
        this.statusCode = statusCode;
    }

    public int getStatusCode() {
        return statusCode;
    }
}

File: tusow-service/src/main/kotlin/it/unibo/coordination/tusow/exceptions/HttpError.kt
package it.unibo.coordination.tusow.exceptions

open class HttpError : RuntimeException {
    val statusCode: Int

    constructor(statusCode: Int) {
        this.statusCode = statusCode
    }

    constructor(statusCode: Int, message: String?) : super(message) {
        this.statusCode = statusCode
    }

    constructor(statusCode: Int, message: String?, cause: Throwable?) : super(message, cause) {
        this.statusCode = statusCode
    }

}